shell_automaton/src/service/storage_service.rs

```diff
@@ -227,6 +228,55 @@ impl fmt::Debug for StorageServiceDefault {
     }
 }
 
+fn find_block_matching_context_storage_impl(
+    head_block_storage: BlockHeaderWithHash,
```

Comment:
Name is a bit confusing. Maybe it should be `head_block` or `head_block_from_storage`?

Reply:
`find_block_with_context_hash` ?

Reply:
Ok, I will rename and add some documentation